# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/ipc/RDDParent.cpp
# Commit: 493b443954fe
# Full Hash: 493b443954fe15f7b542ba14671f25e5f8531dff
# Author: Jed Davis <jld@mozilla.com>
# Date: 2019-02-28 10:55:23
# Regressor Bug: 1506291
# File Overlap Count: 1
# Description:
#   Bug 1506291 - Add Linux sandboxing for the RDD (media decoder) process. r=gcp,mjf,flod
#   
#   The seccomp-bpf policy is currently just the "common" policy with no
#   additions (but with the fixes in bug 1511560 to enable shared memory
#   creation).  The file broker policy allows shared memory creation and
# ==============================================================================

diff -r bf58d8320f5a -r 493b443954fe dom/media/ipc/RDDParent.cpp
--- a/dom/media/ipc/RDDParent.cpp	Mon Feb 25 16:20:50 2019 +0000
+++ b/dom/media/ipc/RDDParent.cpp	Wed Feb 27 20:14:54 2019 +0000
@@ -19,6 +19,10 @@
 #include "mozilla/ipc/CrashReporterClient.h"
 #include "mozilla/ipc/ProcessChild.h"
 
+#if defined(XP_LINUX) && defined(MOZ_SANDBOX)
+#  include "mozilla/Sandbox.h"
+#endif
+
 #ifdef MOZ_GECKO_PROFILER
 #  include "ChildProfilerController.h"
 #endif
@@ -123,12 +127,19 @@
 }
 #endif
 
-mozilla::ipc::IPCResult RDDParent::RecvInit() {
+mozilla::ipc::IPCResult RDDParent::RecvInit(const MaybeFileDesc& aBrokerFd) {
   Unused << SendInitComplete();
-
-#if defined(XP_MACOSX) && defined(MOZ_SANDBOX)
+#if defined(MOZ_SANDBOX)
+#  if defined(XP_MACOSX)
   StartRDDMacSandbox();
-#endif
+#  elif defined(XP_LINUX)
+  int fd = -1;
+  if (aBrokerFd.type() == MaybeFileDesc::TFileDescriptor) {
+    fd = aBrokerFd.get_FileDescriptor().ClonePlatformHandle().release();
+  }
+  SetRemoteDataDecoderSandbox(fd);
+#  endif  // XP_MACOSX/XP_LINUX
+#endif    // MOZ_SANDBOX
 
   return IPC_OK();
 }